### PR TITLE
refactor: complete Phase 2 runtime safety remediation

### PR DESCRIPTION
**Pull request by fbosch**
## Summary
Complete the Phase 2 runtime-safety work by consolidating Playthrough normalization, simplifying encounter and request processing, and adding contract coverage for the affected paths.

## Changes
- normalize persisted and imported Playthrough data through fresh-value migration operations
- make migration steps private and preserve legacy import and storage outcomes
- split encounter, Pokemon, and artist-credit request processing from request handling
- add regression coverage for encounter transitions, migrations, and request contracts

## Risk
Persisted and imported run data now flows through one validated normalization interface. Coverage protects legacy repair, current-shape idempotence, malformed-data handling, and classic-mode preservation. Deceased encounters now clear matching team slots.

## Testing
- `pnpm type-check`
- `pnpm test:run`
- `pnpm validate`
- `pnpm quality:graph:json`

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved playthrough loading and import handling for legacy and malformed data.
  * Ensured deceased encounter Pokémon are removed from active teams.
  * Preserved encounter team updates, fusion statuses, and artwork variant behavior.
  * Improved Pokémon API query validation and filtering.
  * Maintained reliable Pokémon and artist API responses, including error handling and CORS behavior.

* **Tests**
  * Added expanded coverage for API responses, encounter updates, and playthrough normalization.

### CODE REVIEW

File: docs/FALLOW_REMEDIATION_PLAN.md [hunk 1]
```diff
@@ -50,11 +50,13 @@ Purpose: remove dependency-cycle and state-transition risks before broader clean
...
+- [x] Add outcome-focused tests for encounter CRUD state changes, including duplicate catches, nickname requirements, team placement, and death handling.
+- [x] Simplify `updateEncounter`, artwork variants, and migration paths while preserving run-state invariants.
+- [x] Inventory high-complexity API processors by request validation, decisions, side effects, and response behavior.
+- [x] Refactor API processors one request path at a time, with success, invalid-input, and downstream-failure coverage.
+
+Phase 2 API inventory: `api/encounters` validates static datasets, merges and caches route data, and returns a sorted schema-validated response; `api/pokemon` validates query input, filters static data, and returns cache/security headers; `api/sprite/artists` validates an ID, fetches FusionDex HTML, extracts credits, and preserves upstream/error status behavior. The routes now isolate their high-decision processing from HTTP orchestration and have direct request-path coverage. `api/sprite/variants` was inventoried as I/O-heavy but has no top-complexity finding; its request, probe, cache, and response contract remains unchanged.
```
[the remark this answers is not in the document]
Comment (fbosch) [reply docs/FALLOW_REMEDIATION_PLAN.md:59]: Updated the inventory to state that `id` is required and nonempty. The route does not validate a sprite-ID grammar.

File: src/app/api/pokemon/route.ts [hunk 2]
```diff
@@ -10,6 +10,7 @@ const QuerySchema = z.object({
   type: z.string().optional(), // Filter by type
   limit: z
     .string()
+    .regex(/^\d+$/)
     .transform((val) => parseInt(val, 10))
```
[the remark this answers is not in the document]
Comment (fbosch) [reply src/app/api/pokemon/route.ts:14]: Added schema validation for comma-separated numeric IDs and safe-integer limits, with regression coverage for malformed IDs and unsafe limits.

File: src/stores/playthroughs/encounters/status.ts [hunk 3]
```diff
@@ -92,6 +95,12 @@ export const markEncounterAsDeceased = async (
     return;
   }
 
+  removeTeamMembersWithPokemon(
+    [encounterAfter.head?.uid, encounterAfter.body?.uid].filter(
+      (uid): uid is string => uid != null,
+    ),
+  );
```
[the remark this answers is not in the document]
Comment (fbosch) [reply src/stores/playthroughs/encounters/status.ts:102]: Ensured deceased encounters clean matching team members even on repeated calls, with idempotence regression coverage.

**Comment by fbosch** [comment]
@codex review
